### PR TITLE
[WEBSITE] make dropdowns remain openable + remove gap on top of mega menus

### PR DESCRIPTION
**Pull request by Guillaume-gdi**
This commit allows to reopen submenus and mega menus if one of them was
opened when scrolling on a page of a website.

Steps to reproduce the bug fixed by this commit.
- Have a submenu or a mega menu in the navbar
- Open a dropdown of the navbar
- Scroll down

=> It is no longer possible to open the dropdown.

This commit fixes this issue and adds a test for this flow.

---

Since the upgrade to Bootstrap 5 and especially since this [BS5 commit],
there is a gap between the mega menu and the navbar when the user opens
a mega menu. This commit removes that gap.

Steps to reproduce the bug fixed by this commit:
- Have a mega menu on your website.
- Drop a block of a dark color at the top of the page (it helps to see
the problem).

=> When you open the mega menu, you can see a small piece of the block
of dark color between the navbar and the mega menu.

Technical explanation:
This [BS5 commit] introduces a new css rule that adds a
`margin-top: 0.125rem;` on the `.dropdown-menu[data-bs-popper]`. However
when [mega menus were introduced], another css rule prevents having a gap
between the nav and the mega menu (`margin-top: 0;` on `.o_mega_menu`).
This commit makes sure that it will be a `margin-top` of 0 by making the
property more important.

BEFORE:
image
AFTER:
image

[BS5 commit]: https://github.com/odoo/odoo/commit/c48f57ea2538ad51e00ac27d58f8e191781444f3
[mega menus were introduced]: https://github.com/odoo/odoo/commit/1345702258adbfbee0d780dc22e552395e6d1df7

---

task-3133137
opw-3226013

### CODE REVIEW

**Comment by Guillaume-gdi** [comment]
@qsm-odoo This opw PR is ready for your review :heavy_check_mark:

**Comment by qsm-odoo** [comment]
@robodoo delegate=loco-odoo

**Comment by loco-odoo** [comment]
@robodoo rebase-ff r+

**Comment by loco-odoo** [comment]
@robodoo r+

File: addons/website/static/tests/tours/edit_menus.js [hunk 1]
```diff
@@ -174,6 +176,17 @@ wTourUtils.registerWebsitePreviewTour('edit_menus', {
...
+    {
+        content: "Drag Mega at the top",
+        trigger: '.oe_menu_editor li:contains("Megaaaaa!") > .ui-sortable-handle',
+        run: "drag_move_and_drop [0,0]@.oe_menu_editor li:contains('Home') > .ui-sortable-handle => .oe_menu_editor li",
+    },
+    {
+        content: "Wait for drop",
+        trigger: '.oe_menu_editor:first-child:contains("Megaaaaa!")',
+        run: () => {}, // It's a check.
+    },
```
Comment (Guillaume-gdi) [inline addons/website/static/tests/tours/edit_menus.js:189]: @loco-odoo I had to add this in order to prevent the mega menu to be in the + menu

**Comment by loco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**